loki/analyse/util_polyhedron.py

```diff
@@ -0,0 +1,328 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+import numpy as np
+
+from loki.ir import Loop
+
+from loki.expression import (
+    symbols as sym,
+    FindVariables,
+    simplify,
+    is_constant,
+    accumulate_polynomial_terms,
+)
+from loki.tools import as_tuple
+
+__all__ = ["Polyhedron"]
+
+
+class Polyhedron:
+    """
+    Halfspace representation of a (convex) polyhedron.
+
+    A polyhedron `P c R^d` is described by a set of inequalities, in matrix form
+    ```
+    P = { x=[x1,...,xd]^T c R^d | Ax <= b }
+    ```
+    with n-by-d matrix `A` and d-dimensional right hand side `b`.
+
+    In loop transformations, polyhedrons are used to represent iteration spaces of
+    d-deep loop nests.
+
+    :param np.array A: the representation matrix A.
+    :param np.array b: the right hand-side vector b.
+    :param list variables: list of variables representing the dimensions in the polyhedron.
```

Comment:
If you don't mind, change this to the up-to-date docstring format:

```suggestion
    Parameters
    ----------
    A : numpy.array
        The representation matrix A.
    b : numpy.array
        The right hand-side vector b.
    variables : list, optional
        List of variables representing the dimensions in the polyhedron.
```

Reply:
eb8bb06d5ac8af02e35d02a8b9386c1bb7a8eb9b and eb8bb06d5ac8af02e35d02a8b9386c1bb7a8eb9b